looks like upstream enabled a pytest equivalent of -Werror without testing too much
--- a/setup.cfg
+++ b/setup.cfg
@@ -29,6 +29,7 @@
     ignore:unclosed <(socket\.socket|ssl\.SSLSocket) .*:ResourceWarning
     ignore:ssl\.wrap_socket\(\) is deprecated, use SSLContext\.wrap_socket\(\):DeprecationWarning
     ignore:ssl\.PROTOCOL_TLS is deprecated:DeprecationWarning
+    ignore:The 'asyncio_mode' default value will change:DeprecationWarning
 
 [coverage:run]
 omit = venv/*, httpcore/_sync/*
